search/src/Filtering.h

```diff
@@ -9,6 +9,7 @@
 #ifndef FILTERING_H_
 #define FILTERING_H_
 
+#include <queue>
```

Comment:
maybe I'm missing something, but I don't think we're using a queue anywhere in these changes, is it necessary for anything else?

Reply:
Good catch. That is a mistake. I originally added it, thinking I would use a queue like the python approach. But it turns out that it is easier to use a sorted array multiple times and just track indices. So I never ended up using the queue. I've removed the include.